runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/content/ByteStream.kt

```diff
@@ -106,3 +108,92 @@ public fun ByteStream.cancel() {
         is ByteStream.SourceStream -> stream.readFrom().close()
     }
 }
+
+/**
+ * Return a [Flow] that consumes the underlying [ByteStream] when collected.
+ *
+ * @param bufferSizeHint the size of the buffers to emit from the flow. All buffers emitted
+ * will be of this size except for the last one which may be less than the requested buffer size.
+ * This parameter has no effect for the [ByteStream.Buffer] variant. The emitted [ByteArray]
+ * will be whatever size the in-memory buffer already is in that case.
+ */
+public fun ByteStream.toFlow(bufferSizeHint: Long = 8192): Flow<ByteArray> = when (this) {
+    is ByteStream.Buffer -> flowOf(bytes())
+    is ByteStream.ChannelStream -> readFrom().toFlow(bufferSizeHint)
+    is ByteStream.SourceStream -> readFrom().toFlow(bufferSizeHint).flowOn(SdkDispatchers.IO)
+}
```

Comment:
It is except for `ByteStream.Buffer`, I went back and forth on the naming. I also considered making it `nullable` and just emitting buffers without the chunking when not set. Wasn't really sure how customers will want to use this or what they'd expect (controlling the chunk size seemed reasonable though 🤷 )

Reply:
Could we make it default to `null` *just* for the `ByteStream.Buffer` variant and make users opt-in for chunking? Then we can remove the `hint` naming. I think that maximizes clarity while also preserving a sensible default (8192) for the other `ByteStream` variants.

Reply:
> Could we make it default to null just for the ByteStream.Buffer variant 

No, the extension method is on the parent `ByteStream` class.

I think I agree with Ian here that we can just call it `bufferSize` since it's behavior is documented.